scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py

```diff
@@ -22,184 +22,321 @@
-        sys.stderr.write("error(%d): %s\n" %(self._rc, self._desc))
 
     def __str__(self):
-        if self._rc == 0: return self._desc
-        return "error(%d): %s" %(self._rc, self._desc)
+        return f"error[{self._rc}]: {self._desc}"
 
 
-def execute_command(cmd, timeout_secs):
-
+def execute_command(self, cmd, timeout_secs = TIMEOUT_SECS, verbose = False):
     ps = subprocess.Popen(cmd, stdin=subprocess.PIPE,
                           stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
                           shell=True)
     stdout, stderr = ps.communicate(timeout=timeout_secs);
     stdout = str(stdout, 'utf-8')
-    sys.stdout.write(f"[CMD] {cmd}\n")
-    sys.stdout.write(f"[OUT]\n{stdout}\n")
-    sys.stdout.write(f"[RET] {ps.returncode}\n")
+    if self._debug or verbose:
+        sys.stdout.write(f"[CMD] {cmd}\n")
+        sys.stdout.write(f"[OUT]\n{stdout}\n")
+        sys.stdout.write(f"[RET] {ps.returncode}\n")
     if ps.returncode != 0:
-        raise MotrError(ps.returncode, f"{cmd} command failed")
+        raise MotrError(ps.returncode, f"\"{cmd}\" command execution failed")
     return stdout, ps.returncode
 
+def check_type(var, vtype, msg):
+    if not isinstance(var, vtype):
+        raise MotrError(errno.EINVAL, f"Invalid {msg} type. Expected: {vtype}")
+
+
 def get_current_node(self):
+    """Get current node name using machine-id."""
     cmd = "cat /etc/machine-id"
-    machine_id = execute_command(cmd, TIMEOUT_SECS)
+    machine_id = execute_command(self, cmd)
     machine_id = machine_id[0].split('\n')[0]
-    return Conf.get(self._index, 'cluster>server_nodes')[machine_id]
 
-def restart_services(services):
+    check_type(machine_id, str, "machine-id")
+    if len(machine_id) != MACHINE_ID_LEN:
+        raise MotrError(errno.EINVAL, "Invalid machine-id length."
+                        f" Expected: {MACHINE_ID_LEN}"
+                        f" Actual: {len(machine_id)}")
+
+    try:
+        current_node = Conf.get(self._index, 'cluster>server_nodes')[machine_id]
+    except:
+        raise MotrError(errno.EINVAL, "Current node not found")
+
+    check_type(current_node, str, "current node")
+    return current_node
+
+
+def restart_services(self, services):
     for service in services:
-        cmd = "service {} stop".format(service)
-        execute_command(cmd, TIMEOUT_SECS)
-        cmd = "service {} start".format(service)
-        execute_command(cmd, TIMEOUT_SECS)
-        cmd = "service {} status".format(service)
-        execute_command(cmd, TIMEOUT_SECS)
+        sys.stdout.write(f"Restarting {service} service\n")
+        cmd = f"systemctl stop {service}"
+        execute_command(self, cmd)
+        cmd = f"systemctl start {service}"
+        execute_command(self, cmd)
+        cmd = f"systemctl status {service}"
+        execute_command(self, cmd)
 
 def validate_file(file):
     if not os.path.exists(file):
         raise MotrError(errno.ENOENT, "{} not exist".format(file))
```

Comment:
Here, please use python f-string.

If F-String is used elsewhere in this python script, it's better to made them all use the same style.